plugins/modules/podman_generate_systemd.py

```diff
@@ -0,0 +1,572 @@
+    if separator:
+        command_options.append(
+            '--separator={separator}'.format(
+                separator=separator,
+            ),
+        )
+
+    #  No-header option
+    if module.params['no_header']:
+        command_options.append('--no-header')
+
+    #  After option (only for Podman 4.0.0 and above)
+    after = module.params['after']
+    if after:
+        for item in after:
+            command_options.append(
+                '--after={item}'.format(
+                    item=item,
+                ),
+            )
+
+    #  Wants option (only for Podman 4.0.0 and above)
+    wants = module.params['wants']
+    if wants:
+        for item in wants:
+            command_options.append(
+                '--wants={item}'.format(
+                    item=item,
+                )
+            )
+
+    #  Requires option (only for Podman 4.0.0 and above)
+    requires = module.params['requires']
+    if requires:
+        for item in requires:
+            command_options.append(
+                '--requires={item}'.format(
+                    item=item,
+                ),
+            )
+
+    # Environment variables (only for Podman 4.3.0 and above)
+    environment_variables = module.params['env']
+    if environment_variables:
+        for env_var_name, env_var_value in environment_variables.items():
+            command_options.append(
+                "-e='{env_var_name}={env_var_value}'".format(
+                    env_var_name=env_var_name,
+                    env_var_value=env_var_value,
+                ),
+            )
+
+    #   Set output format, of podman command, to json
+    command_options.extend(['--format', 'json'])
+
+    #  Full command build, with option included
+    #   Base of the command
+    command = [
+        module.params['executable'], 'generate', 'systemd',
+    ]
+    #   Add the options to the commande
+    command.extend(command_options)
+    #   Add pod or container name to the command
+    command.append(module.params['name'])
+    #   Build the string version of the command, only for module return
+    command_str = ' '.join(command)
+
+    # Run the podman command to generated systemd .service unit(s) content
+    return_code, stdout, stderr = module.run_command(command)
+
+    # In case of error in running the command
+    if return_code != 0:
+        # Print informations about the error and return and empty dictionary
+        message = 'Error generating systemd .service unit(s).'
+        message += ' Command executed: {command_str}'
+        message += ' Command returned with code: {return_code}.'
+        message += ' Error message: {stderr}.'
+        module.fail_json(
+            msg=message.format(
+                command_str=command_str,
+                return_code=return_code,
+                stderr=stderr,
+            ),
+            changed=changed,
+            systemd_units={},
+            podman_command=command_str,
```

Comment:
This should be removed, `fail_json` accepts only one arg `msg`

Reply:
I checked the source code of `fail_json()`: [Link](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/basic.py#L1516)

This method accept other arguments and will return them as a JSON dictionary at the output of the module. Like `exit_json()` will do.

You can found un example here: [systemd_service module](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/systemd_service.py#L563)

Returning `changed`, `systemd_units` and `podman_command` is necessary because you can call an Ansible module with an option to ignore errors and store the output data in an Ansible variable.
For this case, `podman_generate_systemd` module need to return these 3 values so users can use them.

Reply:
Great, thanks for clarification!